libs/platform/src/lib/components/list/list-item-base.ts

```diff
@@ -0,0 +1,240 @@
+import {
+    ElementRef, HostBinding, Input, ChangeDetectorRef, EventEmitter,
+    Output, HostListener, ViewChild, AfterViewChecked, AfterViewInit
+} from '@angular/core';
+import { CheckboxComponent, RadioButtonComponent } from '@fundamental-ngx/core';
+import { BaseComponent } from '../base';
+import { ENTER, SPACE } from '@angular/cdk/keycodes';
+
+/**
+ * This class contains common properties used across list Item components.
+ * this can be extended to reduce the code duplication across list Item components.
+ */
+export class ListItemBase extends BaseComponent implements AfterViewChecked, AfterViewInit {
```

Comment:
We should rename this `BaseListItem` since we already have established a naming convention of Base classes with `BaseInput`.

Also I feel that there is too much code in this base class. There are elements like "navigation arrow", "byline" and "secondary icon" which appear to be particular to the StandardListItem - I don't see these elements when I look at the specs for ObjectList or ActionList.

Because you only have one List type component, I feel you should put a majority of this code into the StandardListItemComponent. When you work on building a second List component, only then would I extract out the common properties and functions and put them in the Base component.

Reply:
- Name changes are done

- Display list item- uses navigation and secondary icon, along with standard list item.

- Byline- Will update it next commit, as per my understanding object list item has byline feature will check with Manju and styles team on it and will update accordingly

Reply:
Done